Journaley/Utilities/PostMarkdownParser.cs

```diff
@@ -0,0 +1,109 @@
+﻿namespace Journaley.Utilities
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Text.RegularExpressions;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Helper class for fixing MarkdownDeep parsed HTML.
+    /// </summary>
+    public class PostMarkdownParser
+    {
+        /// <summary>
+        /// Perform fixes after MarkdownDeep parsing for publishing
+        /// to Journaley.
+        /// - automatically adds break tags on single line breaks.
+        /// - adds strikethrough style support.
+        /// </summary>
+        /// <param name="formattedString">Formatted HTML string</param>
+        /// <returns>Properly formatted HTML string for publishing.</returns>
+        public static string PostMarkdown(string formattedString)
+        {
+            // First convert all code blocks using <p> into <pre>
+            // which is used in almost all implementations of
+            // Markdown, even in the original Markdown specs.
+            formattedString = Regex.Replace(formattedString, @"<p><code>", "<pre><code>", RegexOptions.Multiline);
+            formattedString = Regex.Replace(formattedString, @"</code></p>", "</code></pre>", RegexOptions.Multiline);
+
+            StringBuilder builder = new System.Text.StringBuilder();
+            StringBuilder paragraphBuilder = new System.Text.StringBuilder();
+
+            // 0 - stumbles upon the beginning of a <p> tag/usual parsing.
+            // 1 - stumbles upon the end of </p> tag.
+            // 2 - stumbles upon a <pre> tag.
+            var parseState = 0;
+
+            string newLineString;
```

Comment:
1. Declare this variable when its used in line 86. This makes it easier to understand the code, because the scope of this variable is actually within the `else if (parseState == 0)` block. It is fine for the other variables like `builder`, `paragraphBuilder`, and `parseState`, because their state persists over the loops.
2. Let's rename this to `paragraph`, because it is a better name, it comes from the `paragraphBuilder`, and the name `newLineString` can be misleading because it usually refers to `\r\n` or some other platform dependent newline string.